src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -63,224 +64,217 @@ const defaultProps = {
-                            onFulfill={this.validateAndSubmitForm}
-                            maxLength={CONST.TFA_CODE_LENGTH}
-                            errorText={this.state.formError.twoFactorAuthCode ? this.props.translate(this.state.formError.twoFactorAuthCode) : ''}
-                            hasError={hasError}
-                            autoFocus
-                        />
-                    </View>
-                ) : (
-                    <View style={[styles.mv3]}>
-                        <MagicCodeInput
-                            autoComplete={this.props.autoComplete}
-                            ref={(el) => (this.inputValidateCode = el)}
-                            label={this.props.translate('common.magicCode')}
-                            name="validateCode"
-                            value={this.state.validateCode}
-                            onChangeText={(text) => this.onTextInput(text, 'validateCode')}
-                            onFulfill={this.validateAndSubmitForm}
-                            errorText={this.state.formError.validateCode ? this.props.translate(this.state.formError.validateCode) : ''}
-                            hasError={hasError}
-                            autoFocus
-                        />
-                        <View style={[styles.changeExpensifyLoginLinkContainer]}>
-                            {this.state.linkSent ? (
-                                <Text style={[styles.mt2]}>{this.props.account.message ? this.props.translate(this.props.account.message) : ''}</Text>
-                            ) : (
-                                <PressableWithFeedback
-                                    style={[styles.mt2, resendButtonStyle]}
-                                    onPress={this.resendValidateCode}
-                                    underlayColor={themeColors.componentBG}
-                                    disabled={this.props.network.isOffline}
-                                    hoverDimmingValue={1}
-                                    pressDimmingValue={0.2}
-                                    accessibilityRole="button"
-                                    accessibilityLabel={this.props.translate('validateCodeForm.magicCodeNotReceived')}
-                                >
-                                    <Text style={[styles.link]}>{this.props.translate('validateCodeForm.magicCodeNotReceived')}</Text>
-                                </PressableWithFeedback>
-                            )}
-                        </View>
-                    </View>
-                )}
+        // We don't include props.account here because requiresTwoFactorAuth won't change, and otherwise
+        // we have recursion on error (since props.account changes to include error property)
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.credentials, props.preferredLocale, twoFactorAuthCode, validateCode]);
```

Comment:
```suggestion
    }, [props.account.requiresTwoFactorAuth, props.credentials, props.preferredLocale, twoFactorAuthCode, validateCode]);
```

I don't see why requiresTwoFactorAuth can't change. Also we don't need to include `props.account`, we should just add `props.account.requiresTwoFactorAuth`